sdk/core/azure-core/azure/core/pipeline/policies/_utils.py

```diff
@@ -0,0 +1,95 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import datetime
+import email.utils
+
+class _FixedOffset(datetime.tzinfo):
+    """Fixed offset in minutes east from UTC.
+
+    Copy/pasted from Python doc
+
+    :param int offset: offset in minutes
+    """
+
+    def __init__(self, offset):
+        self.__offset = datetime.timedelta(minutes=offset)
+
+    def utcoffset(self, dt):
+        return self.__offset
+
+    def tzname(self, dt):
+        return str(self.__offset.total_seconds()/3600)
+
+    def __repr__(self):
+        return "<FixedOffset {}>".format(self.tzname(None))
+
+    def dst(self, dt):
+        return datetime.timedelta(0)
+
+def _parse_http_date(text):
+    """Parse a HTTP date format into datetime."""
+    parsed_date = email.utils.parsedate_tz(text)
+    return datetime.datetime(
+        *parsed_date[:6],
+        tzinfo=_FixedOffset(parsed_date[9]/60)
+    )
+
+def _parse_retry_after(retry_after):
+    """Helper to parse Retry-After and get value in seconds.
+
+    :param str retry_after: Retry-After header
+    :rtype: int
+    """
+    try:
+        delay = int(retry_after)
+    except ValueError:
+        # Not an integer? Try HTTP date
+        retry_date = _parse_http_date(retry_after)
+        delay = (retry_date - datetime.datetime.now(retry_date.tzinfo)).total_seconds()
+    if delay < 0:
+        delay = 0
+    return delay
+
+def get_retry_after(response):
```

Comment:
why are some of the names imported out of this module `public` and others `_private`? I would suggest non-underscore names for anything meant to be imported elsewhere (even out of a private module) but regardless to be consistent.

Reply:
I make get_retry_after public because it will be used out of policy. (poller also uses it)

Reply:
I updated the code. Now starting with "_" means this method is only used within the module. Does this make sense?